### PR TITLE
Haar 1264 link admin to existing gen account

### PR DESCRIPTION
**Pull request by olatomoloju**
Linking a new admin account to an existing general account. With some code merges.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/model/UsageType.kt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package uk.gov.justice.digital.hmpps.manageusersapi.model
+
+enum class UsageType {
+  GENERAL, ADMIN;
+}
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/model/UsageType.kt:5]: We already have a PrisonUsageType, so best to reuse that one

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/prison/UserController.kt [hunk 2]
```diff
@@ -146,6 +149,63 @@ class UserController(
     createUserRequest: CreateUserRequest,
   ) = NewPrisonUserDto.fromDomain(prisonUserService.createUser(createUserRequest))
 
+  @PostMapping("/linkedprisonusers/admin", produces = [MediaType.APPLICATION_JSON_VALUE])
+  @PreAuthorize("hasRole('ROLE_CREATE_USER')")
+  @ResponseStatus(HttpStatus.CREATED)
+  @Operation(
+    summary = "Create a Linked DPS Admin user",
+    description = "Creates a specific Linked DPS user. Requires role ROLE_CREATE_USER",
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/prison/UserController.kt:157]: Wording doesn't feel quite right - possibly "Link an Admin User to an existing General Account" would be better

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/prison/UserController.kt [hunk 3]
```diff
@@ -146,6 +149,63 @@ class UserController(
...
+    security = [SecurityRequirement(name = "ROLE_CREATE_USER")],
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Create a Linked DPS Admin user",
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/prison/UserController.kt:162]: This is really "Admin User linked to General User"

File: src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/prison/UserControllerIntTest.kt [hunk 4]
```diff
@@ -517,6 +626,105 @@ class UserControllerIntTest : IntegrationTestBase() {
     }
   }
 
+  @Nested
+  inner class CreateLinkedAdminUserError {
```
Comment (AlTurner-MOJ) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/prison/UserControllerIntTest.kt:630]: Please discuss, but I don't think all of these error integration test cases are necessary. In the event of a remote call exception we get a WebClientResponseException back. The handler for this basically replicates the error response i.e. so the same error response is given by manage-users-api. So I think only one error test case should suffice. You also don't need to put this in a separate nested inner class.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/integration/wiremock/NomisApiMockServer.kt [hunk 5]
```diff
@@ -175,6 +243,63 @@ class NomisApiMockServer : WireMockServer(WIREMOCK_PORT) {
     )
   }
 
+  fun stubCreateLinkedCentralAdminUserDuplicateConflict() {
```
[the remark this answers is not in the document]
Comment (olatomoloju) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/integration/wiremock/NomisApiMockServer.kt]: Non for Nomis, but I have created one.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/integration/wiremock/NomisApiMockServer.kt [hunk 6]
```diff
@@ -175,6 +243,63 @@ class NomisApiMockServer : WireMockServer(WIREMOCK_PORT) {
...
+                "userMessage": "User already exists: User TEST_USER_ADM already exists",
+                "developerMessage": "User TEST_USER_ADM already exists"
+               }
+              """.trimIndent(),
+            ),
+        ),
+    )
+  }
+
+  fun stubCreateLinkedCentralAdminWhenGeneralUserNotFound() {
```
[the remark this answers is not in the document]
Comment (olatomoloju) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/integration/wiremock/NomisApiMockServer.kt]: I have created one.

**Review by AlTurner-MOJ — APPROVED** [review]
lgtm

**Pull request merged**